pkg/file/redundancy/getter/getter.go

```diff
@@ -138,21 +118,30 @@ func (g *decoder) fetch(ctx context.Context, i int, waitForRecovery bool) (err e
 		}
 	}
 
+	// recovery has started, wait for result instead of fetching from the network
+	select {
+	case <-g.initRecovery:
+		return waitRecovery(nil)
+	default:
+	}
+
 	// first time
 	if g.fly(i) {
 
 		fctx, cancel := context.WithTimeout(ctx, g.config.FetchTimeout)
 		defer cancel()
 
-		g.wg.Add(1)
-		go func() {
-			select {
-			case <-fctx.Done(): // local context
-			case <-g.ctx.Done(): // global context
-			}
-			cancel()
-			g.wg.Done()
-		}()
+		// when the recovery is triggered, we can terminate any inflight requests.
+		// we do the extra bool check to not fire an unnecessary goroutine
+		if waitForRecovery {
```

Comment:
there is no need to run this routine for every fetcgh.
creating context and cancel hould be done onky once

Reply:
context is canceled is a separate goroutine when the recovery is initialized...this is how we achieve early termination.
it's safe to call cancel twice in separate goroutines.

Reply:
it would make sense to cancel on